tests/devices/qutrit_mixed/test_qutrit_mixed_simulate.py

```diff
@@ -0,0 +1,735 @@
+
+        import jax
+
+        phi = jax.numpy.array(0.678)
+
+        def f(x):
+            qs = self.get_TRX_quantum_script(x, subspace)
+            return simulate(qs)
+
+        if use_jit:
+            f = jax.jit(f)
+
+        result = f(phi)
+        expected = expected_TRX_circ_expval_values(phi, subspace)
+        assert qml.math.allclose(result, expected)
+
+        g = jax.jacobian(f)(phi)
+        expected = expected_TRX_circ_expval_jacobians(phi, subspace)
+        assert qml.math.allclose(g, expected)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("subspace", [(0, 1), (0, 2)])
+    def test_torch_results_and_backprop(self, subspace):
+        """Tests execution and gradients of a simple circuit with torch."""
+
+        import torch
+
+        phi = torch.tensor(-0.526, requires_grad=True)
+
+        def f(x):
+            qs = self.get_TRX_quantum_script(x, subspace)
+            return simulate(qs)
+
+        result = f(phi)
+        expected = expected_TRX_circ_expval_values(phi.detach().numpy(), subspace)
+        assert qml.math.allclose(result[0], expected[0])
+        assert qml.math.allclose(result[1], expected[1])
+        assert qml.math.allclose(result[2], expected[2])
+        assert qml.math.allclose(result[3], expected[3])
+
+        g = torch.autograd.functional.jacobian(f, phi + 0j)
+        expected = expected_TRX_circ_expval_jacobians(phi.detach().numpy(), subspace)
+        assert qml.math.allclose(g[0], expected[0])
+        assert qml.math.allclose(g[1], expected[1])
+        assert qml.math.allclose(g[2], expected[2])
+        assert qml.math.allclose(g[3], expected[3])
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("subspace", [(0, 1), (0, 2)])
+    def test_tf_results_and_backprop(self, subspace):
+        """Tests execution and gradients of a simple circuit with tensorflow."""
+        import tensorflow as tf
+
+        phi = tf.Variable(4.873)
+
+        with tf.GradientTape(persistent=True) as grad_tape:
+            qs = self.get_TRX_quantum_script(phi, subspace)
+            result = simulate(qs)
+
+        expected = expected_TRX_circ_expval_values(phi, subspace)
+        assert qml.math.allclose(result, expected)
+
+        grad0 = grad_tape.jacobian(result[0], [phi])
+        grad1 = grad_tape.jacobian(result[1], [phi])
+        grad2 = grad_tape.jacobian(result[2], [phi])
+        grad3 = grad_tape.jacobian(result[3], [phi])
+
+        expected = expected_TRX_circ_expval_jacobians(phi, subspace)
+        assert qml.math.allclose(grad0[0], expected[0])
+        assert qml.math.allclose(grad1[0], expected[1])
+        assert qml.math.allclose(grad2[0], expected[2])
+        assert qml.math.allclose(grad3[0], expected[3])
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("op", [qml.TRX(np.pi, 0), qml.QutritBasisState([1], 0)])
+    def test_result_has_correct_interface(self, op):
+        """Test that even if no interface parameters are given, result is correct."""
```

Comment:
I'm not sure what you mean here, because `interface='jax'` is being explicitly passed to `simulate` in this test.

Reply:
This checks that simulate applies that interface and it ends up in the result. Its fairly simple, but I think it is important. I could do it for all of the interfaces too if you think that would be better.